src/Data/Nat/DivMod.agda

```diff
@@ -324,17 +325,17 @@ m*n/m*o≡n/o m n o = helper (<-wellFounded n)
     n / o                         ∎
     where n∸o<n = ∸-monoʳ-< (n≢0⇒n>0 (≢-nonZero⁻¹ o)) n≥o
 
-m*n/o*n≡m/o : ∀ m n o .{{_ : NonZero o}} .{{_ : NonZero (o * n)}} →
+m*n/o*n≡m/o : ∀ m n o .{{_ : NonZero n}} .{{_ : NonZero o}} →
+              let instance _ = m*n≢0 o n in
               m * n / (o * n) ≡ m / o
 m*n/o*n≡m/o m n o = begin-equality
-  m * n / (o * n) ≡⟨ /-congˡ (*-comm m n) ⟩
+  m * n / (o * n) ≡⟨ /-congˡ {{o*n≢0}} (*-comm m n) ⟩
```

Comment:
whereas here you indeed do what I 'expected', i.e. pass it in (though I don't know why you bother naming it)

Reply:
Well, this one is *unbelievably* annoying. The old proof had no problem directly using the in-scope, but irrelevant/unmentioned, instance of `{{NonZero (o * n)}}` to ensure that both the term was well-formed, and that `/-congˡ (*-comm m n)` was well-typed. But the change I made has the typechecker unable to infer the sub-term `(o * n)`, leaving two choices: supply it as a named instantiation of the implicit parameter to that proof, or else (re-)supply the instance as a proxy for that. And this despite the instance required being irrelevant! So maybe it's an Agda bug, maybe it points to the whole venture being misguided, but it's *not* what (I believe) you expected.

Reply:
As for "hand-constructing proofs": no, I disagree with that analysis, too. What I'm doing is choosing an inference rule to preferentially apply, towards Agda constructing the eventual complete proof, based on the search that it will *still* have to do to find suitable instances of the 'atomic' `{{NonZero n}}` and `{{NonZero o}}` hypotheses...

Reply:
Indeed, this may well be an Agda weakness / bug.